Fix notification filtering

MarcoGuiducci opened the pull request:
Filtering of notifications by "title" don't work properly. The query don't provide any result because the condition is wrong. Fixed the parseString method of NotificationServiceFilterParser class to let the notification filtering works well

MarcoGuiducci committed:
Fix notification filtering

Filtering of notifications by "title" don't work properly. The query don't provide any result because the condition is wrong.
Fixed the parseString method of NotificationServiceFilterParser class to let the notification filtering works well

mcop1 commented:
@MarcoGuiducci I know, it is not really part of your pr, but when I activate the checkbox next to the filter, without text, I get an error message:
![grafik](https://user-images.githubusercontent.com/89011527/199706887-fa4de5d8-ce41-477a-9c4a-f2a3fceeb96a.png)
=>
![grafik](https://user-images.githubusercontent.com/89011527/199706657-dc8c814e-7537-45f4-8da6-0549eecf214b.png)

Could you please add an check for the item in the array here https://github.com/pimcore/pimcore/pull/13457/files#diff-50f76d3198e6e87ad4707d5291996fef67e8c300a620ca31adf3fe85f098a031L113 ? 

Thank you for your efforts!

MarcoGuiducci committed:
check for existence of the value parameter

this check permits to avoid errors so that all notifications are shown if no text is typed in the filter

MarcoGuiducci commented:
@mcop1 done in this commit.
[https://github.com/pimcore/pimcore/pull/13457/commits/1f5037d74c05eed840495fc19aa53f1a8f44f31d](https://github.com/pimcore/pimcore/pull/13457/commits/1f5037d74c05eed840495fc19aa53f1a8f44f31d)

jdreesen commented on models/Notification/Service/NotificationServiceFilterParser.php:
```diff
@@ -110,11 +110,11 @@ private function parseString(array $item): array
         $result = null;
         $property = $this->getDbProperty($item);
-        $value = $item[self::KEY_VALUE];
+        $value = array_key_exists(self::KEY_VALUE, $item) ? $item[self::KEY_VALUE] : "";
```
We don't have to consider `null` here, right?
```suggestion
        $value = $item[self::KEY_VALUE] ?? '';
```

mcop1 committed:
Update models/Notification/Service/NotificationServiceFilterParser.php

Co-authored-by: Jacob Dreesen <jacob@hdreesen.de>
